idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/extensions/ExtensionProviderSyntax.kt

```diff
@@ -13,65 +13,179 @@ import com.intellij.openapi.util.ClassExtension
 import com.intellij.psi.PsiElement
 import org.jetbrains.kotlin.diagnostics.Diagnostic
 import org.jetbrains.kotlin.resolve.diagnostics.DiagnosticSuppressor
+import arrow.meta.ide.dsl.editor.icon.IconProviderSyntax
+import arrow.meta.dsl.analysis.AnalysisSyntax
+import arrow.meta.ide.dsl.editor.lineMarker.LineMarkerSyntax
+import arrow.meta.ide.dsl.editor.search.SearchSyntax
+import arrow.meta.ide.dsl.editor.inspection.InspectionSyntax
 
-
+/**
+ * The [ExtensionProvider] phase sits at the core of the main API in IntelliJ's Platform for ExtensionPoints.
+ * ExtensionPoints regardless of their nature are means to interact with the editor.
```

Comment:
nit: `editor`->`ide`? In IntelliJ wording `editor` is the text editor component to modify a (text) file.

Reply:
is there another synonym I can use instead?

Reply:
Perhaps just `IDE` instead of `editor`? IDE is the whole thing, editor a part of it

Reply:
Thank you, I changed it.